pkg/controller/perconaservermongodb/users.go

```diff
@@ -0,0 +1,260 @@
+	for i, replset := range cr.Spec.Replsets {
+		if i > 0 {
+			log.Info("update users: multiple replica sets is not yet supported")
+			return nil
+		}
+
+		matchLabels := map[string]string{
+			"app.kubernetes.io/name":       "percona-server-mongodb",
+			"app.kubernetes.io/instance":   cr.Name,
+			"app.kubernetes.io/replset":    replset.Name,
+			"app.kubernetes.io/managed-by": "percona-server-mongodb-operator",
+			"app.kubernetes.io/part-of":    "percona-server-mongodb",
+		}
+
+		pods := &corev1.PodList{}
+		err := r.client.List(context.TODO(),
+			pods,
+			&client.ListOptions{
+				Namespace:     cr.Namespace,
+				LabelSelector: labels.SelectorFromSet(matchLabels),
+			},
+		)
+		if err != nil {
+			return errors.Wrapf(err, "get pods list for replset %s", replset.Name)
+		}
+		rsAddrs, err := psmdb.GetReplsetAddrs(r.client, cr, replset, pods.Items)
+		if err != nil {
+			return errors.Wrap(err, "get replset addr")
+		}
+		adminPass := string(internalSysSecretObj.Data[envMongoDBUserAdminPassword])
+		adminUser := string(internalSysSecretObj.Data[envMongoDBUserAdminUser])
+		client, err := mongo.Dial(rsAddrs, replset.Name, adminUser, adminPass, true)
+		if err != nil {
+			client, err = mongo.Dial(rsAddrs, replset.Name, adminUser, adminPass, false)
+			if err != nil {
+				return errors.Wrap(err, "dial:")
+			}
+		}
+		defer client.Disconnect(context.TODO())
+
+		for _, user := range users {
+			res := client.Database("admin").RunCommand(context.TODO(), bson.D{{Key: "updateUser", Value: user.Name}, {Key: "pwd", Value: user.Pass}})
+			if res.Err() != nil {
+				return errors.Wrap(res.Err(), "change pass")
+			}
+		}
+	}
+
+	return nil
+}
+
+func sysUsersSecretDataChanged(newHash string, usersSecret *corev1.Secret) (bool, error) {
+	secretData, err := json.Marshal(usersSecret.Data)
+	if err != nil {
+		return false, err
+	}
+	oldHash := sha256Hash(secretData)
+
+	return oldHash != newHash, nil
+}
+
+func sha256Hash(data []byte) string {
+	return fmt.Sprintf("%x", sha256.Sum256(data))
+}
+
+func (r *ReconcilePerconaServerMongoDB) restartPods(cr *api.PerconaServerMongoDB, newSecretDataHash string) error {
+	for _, rs := range cr.Spec.Replsets {
+		sfs := appsv1.StatefulSet{}
+		err := r.client.Get(context.TODO(),
+			types.NamespacedName{
+				Namespace: cr.Namespace,
+				Name:      cr.Name + "-" + rs.Name,
+			},
+			&sfs,
+		)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get stetefulset '%s'", rs.Name)
+		}
+
+		for i := 0; i < int(*sfs.Spec.Replicas); i++ {
+			pod := corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      cr.Name + "-" + rs.Name + "-" + strconv.Itoa(i),
+					Namespace: cr.Namespace,
+				},
+			}
+			err = r.client.Delete(context.TODO(), &pod)
+			if err != nil {
+				return errors.Wrapf(err, "delete pod '%s'", cr.Name+"-"+rs.Name+"-"+strconv.Itoa(int(i)))
+			}
+		}
+	}
+	return nil
+}
```

Comment:
1. Everywhere else pod lists are retrieved by labels: https://github.com/percona/percona-server-mongodb-operator/blob/6c6c2ae9deb324e60cd20c5ef13d35abf56c310e/pkg/controller/perconaservermongodb/psmdb_controller.go#L263-L278

2. I doubt changing the logic from sfs recreation to pods removing (forcing k8s to recreate it). Would be better to understand why sfs recreation doesn't work.

Reply:
bug fixed in https://github.com/percona/percona-server-mongodb-operator/pull/387/commits/3db194680549e61302e26775d35d1b3ea171b96c 
logic added in https://github.com/percona/percona-server-mongodb-operator/pull/387/commits/2b4a4d85a6141b122f94b5fc4ef1996d21a3bf5e